Increase the default PVC size for logstash (#7540)
elastic/cloud-on-k8s: docs/orchestrating-elastic-stack-applications/logstash.asciidoc

[discrete]
== Specifying the volume claim settings

A PersistentVolume called `logstash-data` is created by default.
It maps to `/usr/share/logstash/data` for persistent storage, which is typically used for storage from plugins. 

> By default, the `logstash-data` volume claim is a 1.5Gi volume, using the standard StorageClass of your Kubernetes cluster. 

Reviewer: nit, to be consistent with ``[...] the storage to `2Gi` for [...]`` below
```suggestion
By default, the `logstash-data` volume claim is a `1.5Gi` volume, using the standard StorageClass of your Kubernetes cluster. 
```

[discrete]
== Specifying the volume claim settings

By default, a PersistentVolume called `logstash-data` is created, that maps to `/usr/share/logstash/data` for persistent storage, typically used for storage from plugins. The `logstash-data` volume claim is, by default, a 1.5Gi volume, using the standard StorageClass of your Kubernetes cluster, but can be overridden by adding a `spec.volumeClaimTemplate` section named `logstash-data`.

For production workloads, you should define your own volume claim template with the desired storage capacity and (optionally) the Kubernetes link:https://kubernetes.io/docs/concepts/storage/storage-classes/[storage class] to associate with the persistent volume. To override this volume claim for `data` usages, the name of this volume claim must be `logstash-data`.

> 

Reviewer 2: I think it might be a good idea to add extra information on using persistent volumes, something like:

In particular, when enabling the persistent queue feature, using dead letter queues, or certain plugins that make heavy use of temporary storage, the default volume size will likely be insufficient for production workloads and you should increase the storage capacity, or consider creating  separate volumes for these use cases.

Maybe just before the `Separate storage` section?

            storage: 2Gi
----

> In particular, when enabling the persistent queue (PQ) feature, using dead letter queues (DLQ), or certain plugins that make heavy use of temporary storage, the default volume size will likely be insufficient for production workloads and you should increase the storage capacity, or consider creating separate volumes for these use cases.

Reviewer 3: ```suggestion
The default volume size will likely be insufficient for production workloads, espeically when you are using:

* the persistent queue (PQ) feature
* dead letter queues (DLQ), or 
* {ls} plugins that make heavy use of temporary storage. 

Increase the storage capacity, or consider creating separate volumes for these use cases.
```

> Separate storage can be added by including an additional `spec.volumeClaimTemplate` along with a corresponding `spec.podTemplate.spec.containers.volumeMount` for each requested volume.

Reviewer 3: ```suggestion
You can add separate storage by including an additional `spec.volumeClaimTemplate` along with a corresponding `spec.podTemplate.spec.containers.volumeMount` for each requested volume.
```
